### ISSUE TITLE
Webpack's parser error

### ISSUE THREAD

**Issue by chicoxyzzy** [webpack/webpack#688]
Here is my sample project https://github.com/chicoxyzzy/real-world-react/tree/webpack (notice webpack branch)

When I run webpack I get error

```
ERROR in ./~/core-js/library.js
Module parse failed: /PATH_TO_MY_TEST_APP/node_modules/6to5-loader/index.js?optional=coreAliasing!/PATH_TO_MY_TEST_APP/node_modules/core-js/library.js Line 1149: Unexpected token .
You may need an appropriate loader to handle this file type.
|   // ES6 promises shim
|   // Based on https://github.com/getify/native-promise-only/
|   !(function (_core.Promise, test) {
|     isFunction(_core.Promise) && isFunction(_core.Promise.resolve) && _core.Promise.resolve(test = new _core.Promise(function () {})) == test || (function (asap, DEF) {
|       function isThenable(o) {
 @ ./src/index.js 7:28-54
```

source file comes to Esprima parser already broken. As you can see function parameter has comma in it

**Comment by sirlancelot** [comment]
`!(function (_core.Promise, test) {` You can't have a `.` dot in a function definition's parameters.

**Comment by sokra** [comment]
Looks like a `6to5` bug...

**Comment by chicoxyzzy** [comment]
`6to5` works great with `browserify` (you can check `master` branch of my repo)
also I came here from [6to5's issue](https://github.com/6to5/6to5/issues/377#issuecomment-69695825) where @sebmck said it might be `webpack`'s issue

**Comment by sebmck** [comment]
@sokra The parser error is not coming from 6to5.

**Comment by sokra** [comment]
@sebmck The error isn't, but 6to5 generates invalid javascript

``` js
> to5.transform("!(function(Promise) {}())", {optional: ["coreAliasing"]}).code
```

returns

``` js
"use strict";

var _interopRequire = function (obj) {
  return obj && (obj["default"] || obj);
};

var _core = _interopRequire(require("core-js/library"));

!(function (_core.Promise) {})();
```

The last line is invalid javascript.

**Comment by sebmck** [comment]
@sokra Oh right, I didn't read the syntax error properly, sorry! Fixed in 2.12.2. We weren't taking into account function parameters when checking if an identifier was referenced.

**Comment by sebmck** [comment]
@sokra Oh right, I didn't read the syntax error properly, sorry! Fixed in 2.12.2. We weren't taking into account function parameters when checking if an identifier was referenced.

**Comment by sokra** [comment]
:smile:
